pkg/ccl/crosscluster/logical/lww_row_processor.go

```diff
@@ -192,17 +190,53 @@ func (q *queryBuffer) ApplierQueryForRow(row cdcevent.Row) (queryBuilder, error)
 	return applierQueryBuilder, nil
 }
 
+type sqlProcessorTableConfig struct {
+	srcDesc   catalog.TableDescriptor
+	dstDBName string
+	dstFnName string
+}
+
+// validateTableConfigs validates assumptions made in the current
+// implementation. Most of these can be lifted in the future but are
+// disallowed for now to reduce churn in this first cut.
+func validateTableConfigs(configs map[descpb.ID]sqlProcessorTableConfig) error {
+	var (
+		first       = true
+		dbName      string
+		hasFunction bool
+	)
+	for _, conf := range configs {
+		if first {
+			dbName = conf.dstDBName
+			hasFunction = conf.dstFnName != ""
+			first = false
+		}
+		if dbName != conf.dstDBName {
+			return errors.Errorf("cross-database replication job is not allowed")
+		}
+		if hasFunction != (conf.dstFnName != "") {
+			return errors.Errorf("function must be specified for all tables")
```

Comment:
huh... why?

Reply:
Actually though, the code to do this is thinner than I thought. Will push it to this PR.